src/content/blog/2022-08-03-new-config-system-part-1.md

```diff
@@ -0,0 +1,130 @@
+
+### Personal configs
+
+The next layer of complexity was added when people requested that we add the abilty to have a personal config file at `~/.eslintrc`. So we added an additional check: if we didn't find a config file in the ancestry of the file location, then we would automatically look for a personal config file.
+
+### Multiple config file formats
+
+As part of a refactor, we discovered that it would be trivial to allow different config file formats. Instead of forcing everyone to use a nonstandard `.eslintrc` file, we could formalize the JSON format as `.eslintrc.json` and also add support for YAML (`.eslintrc.yml` or `.eslintrc.yaml`) and JavaScript (`.eslintrc.js`). For backwards compatibility we continued to support `.eslintrc` because it was a trivial amount of code to keep around.
+
+This also turned out, in hindsight, to not be a great idea. Adding a JavaScript config file format created an incompatibility between it and the non-JS formats: any JavaScript object could be passed into the config and available in rules. Because we didn't properly validate the config to exactly match the non-JS formats, we ended up with some rules requiring regular expression objects to be passed in to be properly configured. While this could work in the JS config file format, the rules could not be properly configured in non-JS config files. Unfortunately, because plugin rules depended on this functionality, we couldn't go back and fix it without breaking things.
+
+### Shareable configs and dependencies
+
+Perhaps the biggest problem we faced early on was when npm decided to [stop installing peer dependencies in v3](https://github.com/npm/npm/issues/6565#issuecomment-74971689). Prior to this point, we had recommended that shareable configs include any plugins they depended on as peer dependencies rather than regular dependencies. This was a quirk of the way that `extends` was implemented: using `require()`.
+
+Because shareable configs were data-only and couldn't directly reference Node.js dependencies, `require()` would not automatically load direct dependencies into the path for ESLint to resolve them. Peer dependencies, on the other hand, worked perfectly by just using `require()` because those were installed in a location where the normal package lookup worked.
+
+When npm v3 stopped installing peer dependencies by default, all of the shared configs relying on this behavior stopped working correctly. There is a [long-running issue](https://github.com/eslint/eslint/issues/3458) requesting that shareable configs be allowed to use dependencies directly, but the architecture of eslintrc just didn't allow for it. We would have essentially had to recreate the entire `require()` functionality inside of ESLint to work around the way shareable configs were designed. We recommended that shareable configs create a post-install script to install their peer dependencies instead. Not ideal by any stretch of the imagination.
```

Comment:
I would also mentioned that we've tried to get help from node team directly by requesting modifications for `require.resolve` functionality, but it was never done.

Reply:
It was eventually done! :tada: